localstack/services/cloudformation/models/ec2.py

```diff
@@ -8,6 +8,8 @@
 from localstack.utils.aws import aws_stack
 from localstack.utils.strings import str_to_bool
 
+TAG_KEY_LOGICAL_ID = "aws:cloudformation:logical-id"
```

Comment:
Is there something that explicitly breaks on this missing? Because I'm not sure about only implementing it here when it should actually be included nearly everywhere :thinking:

Reply:
That's a great catch. This change actually hints at a general limitation we have in our approach (as you're well aware) - the way how we detect the resource state and the fact that we're allowing templates to be deployed on top of existing resources.

With the current implementation of our resource classes, in combination with the [change above](https://github.com/localstack/localstack/pull/7711/files#diff-95b30e7b954acaf70be05be62cb7404d8fef4ddf7538b924e12724f73f369bf0R1434-R1435), the `fetch_state` method can sometimes return false positives - i.e., report that the resource already exists when in fact it doesn't exist yet.

This issue in particular occurs if we have multiple instances of resource types whose deployment state cannot easily be determined based on the resource properties, e.g., consider the example below (we have a [similar example in our tests](https://github.com/localstack/localstack/blob/master/tests/integration/templates/template36.yaml#L15-L40)):
```
  routeTable1:
    Type: AWS::EC2::RouteTable
    Properties:
      VpcId: vpc-12345
  routeTable2:
    Type: AWS::EC2::RouteTable
    Properties:
      VpcId: vpc-12345
```

I've simplified the implementation - now the state lookup detects whether `PhysicalResourceId` has already been set (using the `result_handler` hook) and only then returns the resource details. This is quite a bit cleaner, and also should give us better parity with AWS (no additional tags). 👌